File: packages/functions/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/functions/classes/options/CallByNameOptions.java
package io.capawesome.capacitorjs.plugins.firebase.functions.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class CallByNameOptions extends CallOptions {

    @NonNull
    private String name;

    @Nullable
    private String region;

    public CallByNameOptions(@NonNull String name, @Nullable String region,  Object data) {
        super(data);
        this.name = name;
        this.region = region;
    }

    @NonNull
    public String getName() {
        return name;
    }

    @Nullable
    public String getRegion() {
        return region;
    }
}

File: packages/functions/ios/Plugin/Classes/Options/CallByNameOptions.swift
import Foundation

@objc public class CallByNameOptions: CallOptions {
    private var name: String
    private var region: String?

    init(name: String, region: String?, data: Any?, timeout: Int?) {
        self.name = name
        self.region = region
        super.init(data: data, timeout: timeout)
    }

    func getName() -> String {
        return name
    }

    func getRegion() -> String? {
        return region
    }
}
